src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7420,7 +7420,38 @@ public StatementSyntax ParseStatement()
         }
 
         private StatementSyntax ParsePossiblyAttributedStatement()
-            => ParseStatementCore(ParseAttributeDeclarations(inExpressionContext: true), isGlobal: false);
+            => ParseStatementCore(ParseStatementAttributeDeclarations(), isGlobal: false);
+
+        private SyntaxList<AttributeListSyntax> ParseStatementAttributeDeclarations()
+        {
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return default;
+
+            // See if we should treat this as a collection expression.  At the top-level or statement-level, this should
+            // only be considered a collection if followed by a `.`, `?` or `!` (indicating it's a value, not an
+            // attribute).
+            var resetPoint = GetResetPoint();
+
+            // Grab the first part as a collection expression.
+            var expression = (ExpressionSyntax)ParseCollectionCreationExpression();
+
+            // continue consuming element access expressions for `[x][y]...`.  We have to determine if this is a
+            // collection expression being indexed into, or if its a sequence of attributes.
+            while (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
+                expression = _syntaxFactory.ElementAccessExpression(expression, ParseBracketedArgumentList());
```

Comment:
>_syntaxFactory.ElementAccessExpression [](http://example.com/codeflow?start=29&length=38)

Do we need to create an `ElementAccessExpression()` or hold onto the `expression`?

Reply:
nope.  removed.